src/components/TodoFormList.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { useGlobalState } from '../CustomHooks/useGlobalState';
+import { useDispatch } from '../CustomHooks/useDispatch';
+import { deleteTodoFromStorage, patchTodoFromStorage } from '../api/todos';
+import classNames from 'classnames';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoFormList: React.FC<Props> = ({ todo }) => {
+  const { id, completed, title } = todo;
+
+  const [isEditTodo, setIsEditTodo] = useState<{
+    [id: number]: boolean;
+  } | null>(null);
+  const [isEditing, setIsEditing] = useState(false);
```

Comment:
user should be able to edit only 1 todo at a time - so just storing its id (number) would be enough